### PR TITLE
refactor: deno_crypto op crate

### PR DESCRIPTION
**Pull request by littledivy**
Moves crypto to its own op_crate

### CODE REVIEW

**Review by ry — COMMENTED** [review]
@littledivy Thanks for this patch - I think this looks good and is on the right track.

I'm wondering why you're working on this? Do you need this yourself - or just thought it'd be a good clean up?

**Comment by littledivy** [comment]
@ry Saw it on the the Roadmap #7915, thought I'd help :)

> finish op crates refactor - tests, remaining web APIs in crates

Crypto is very likely to expand in the near future so just a step towards keeping it clean for contributions.

**Comment by littledivy** [comment]
Would you mind patches for moving most of the ops? – I didn't attempt to refract all of them in this PR as I wanted to know the core team's opinion.

**Comment by bartlomieju** [comment]
> Would you mind patches for moving most of the ops? – I didn't attempt to refract all of them in this PR as I wanted to know the core team's opinion.

Thanks @littledivy but we should hold off on moving more crates - it's not exactly clear yet how we should handle reexports and where to draw crates boundaries; additionally remaining JS runtime code is quite dependent on some code that is specific to `cli`

**Comment by littledivy** [comment]
Should be ready to land :+1:

File: op_crates/crypto/Cargo.toml
Comment (bartlomieju) [inline op_crates/crypto/Cargo.toml:7]: @littledivy please modify `.github/ci.yml` to publish this create similar to `deno_web`

File: op_crates/crypto/Cargo.toml
Comment (bartlomieju) [inline op_crates/crypto/Cargo.toml:18]: It'd be great if we could remove `rand` dependency from `cli/` then - that would require to slightly change `ops/fs/` to use `tempfile` crate instead of `rand` - @littledivy could you look into that?

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: op_crates/crypto/Cargo.toml
Comment (littledivy) [reply op_crates/crypto/Cargo.toml:18]: `crypto.rs` is still dependent on rand - https://github.com/denoland/deno/pull/7956/files#diff-1edd7a8788f2b6027674aee860437562bcd17c1aa7cd0d0ef6c541548f0f2fe1R3-R4

File: op_crates/crypto/Cargo.toml
Comment (littledivy) [reply op_crates/crypto/Cargo.toml]: done

File: op_crates/crypto/Cargo.toml
Comment (bartlomieju) [reply op_crates/crypto/Cargo.toml:18]: @littledivy you're right... I guess we could reexport `rand` from `deno_crypto` then

**Comment by littledivy** [comment]
merged master.

**Comment by bartlomieju** [comment]
@littledivy please, I'm still interested in landing this PR

**Comment by littledivy** [comment]
@bartlomieju Updated. :+1: (sorry for the delay)

**Review by bartlomieju — APPROVED** [review]
LGTM

Thank you @littledivy, hopefully this refactor will kick-off `webcrypto` APIs

**Pull request merged**
